File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        
        if(n <= 1)  return n;
        
        if(n == 2)  return n - 1;
        
        vector<int>dp(4, 0);
        
        dp[0] = 0, dp[1] = 1, dp[2] = 1;
        
        for(int i = 3; i <= n; i++){
            
            dp[3] = dp[0] + dp[1] + dp[2];
            
            dp[0] = dp[1];
            
            dp[1] = dp[2];
            
            dp[2] = dp[3];
            
        }
        
        return dp[3];
        
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        int t0 = 0, t1 = 1, t2 = 1;
        int res = 0;
        
        if(n == 0)  
            return 0;
        
        if(n < 3)
            return 1;
        
        for(int i = 3; i <= n; i++){
            res = t0 + t1 + t2;
            t0 = t1;
            t1 = t2;
            t2 = res;
        }
        
        return res;
    }
}